### ISSUE TITLE
[3.3.5] MMaps Generator shows wrong %

### ISSUE THREAD

**Issue by jackpoz** [TrinityCore/TrinityCore#18832]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
The mmaps generator shows 64% and then shows the message "Finished. MMAPS were built". This makes the % totally useless.
Issue added in https://github.com/TrinityCore/TrinityCore/pull/17798

**Current behaviour:** (Tell us what happens.)
The mmaps generator shows 64% and then shows the message "Finished. MMAPS were built".

**Expected behaviour:** (Tell us what should happen instead.)
The % should show in the range [0,100] % .

**Steps to reproduce the problem:**

1. Delete any previous generated mmaps folder
2. .\mmaps_generator --threads 4
3. See this log showing
>64% [Map 598] Building tile [32,32]
[Map 576] [32,31]:  Building movemap tiles...
[Map 598] [32,32]:  Building movemap tiles...
[Map 598] [32,32]:  Building navmesh tile...
[Map 598] [32,32]:  Adding tile to navmesh...
[Map 598] [32,32]:  Writing to file...
[Map 598] Complete!
[Map 576] [32,31]:  Building navmesh tile...
[Map 576] [32,31]:  Adding tile to navmesh...
[Map 576] [32,31]:  Writing to file...
64% [Map 576] Building tile [33,30]
[Map 576] [33,30]:  Building movemap tiles...
[Map 548] [33,32]:  Building navmesh tile...
[Map 548] [33,32]:  Adding tile to navmesh...
[Map 548] [33,32]:  Writing to file...
64% [Map 548] Building tile [34,30]
[Map 548] [34,30]:  Building movemap tiles...
[Map 576] [33,30]:  Building navmesh tile...
[Map 576] [33,30]:  Adding tile to navmesh...
[Map 576] [33,30]:  Writing to file...
64% [Map 576] Building tile [33,31]
[Map 576] [33,31]:  Building movemap tiles...
64% [Map 548] Building tile [34,31]
[Map 548] [34,31]:  Building movemap tiles...
[Map 576] Complete!
[Map 548] [34,31]:  Building navmesh tile...
[Map 548] [34,31]:  Adding tile to navmesh...
[Map 548] [34,31]:  Writing to file...
64% [Map 548] Building tile [34,32]
[Map 548] [34,32]:  Building movemap tiles...
[Map 548] [34,32]:  Building navmesh tile...
[Map 548] [34,32]:  Adding tile to navmesh...
[Map 548] [34,32]:  Writing to file...
[Map 548] Complete!
Finished. MMAPS were built in 6926419 ms!

**Branch(es):** 3.3.5 / master (Tell us which branch(es) this issue affects.)
3.3.5

**TC rev. hash/commit:**  
a6acd8885a2e0ccb7337e16dd4d7ab10890953c0

**TDB version:**  
no clue

**Operating system:**  
Windows 10 x64

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by jackpoz** [comment]
07593f6c3488666423093c2c32514a0a378da942
>95% [Map 576] Building tile [33,31]
[Map 576] [33,31]:  Building movemap tiles...
[Map 576] Complete!
Finished. MMAPS were built in 8857177 ms!
PS D:\program files\World of Warcraft 3.3.5a>

**Comment by Rochet2** [comment]
Tested building mmaps for mapid 547 only and the result was that there was 5750 tiles discovered, but less than 10 actually built. The percentage was at 0 all the time because 10*100/5750 < 1.  Not all discovered tiles are built but they are a part of the m_totalTiles that is used for the percentage.
Removing this line: https://github.com/TrinityCore/TrinityCore/blob/07593f6c3488666423093c2c32514a0a378da942/src/tools/mmaps_generator/MapBuilder.cpp#L158
seemed to fix the issue for map 547 but I dont think this will fix the whole thing. And did not test building all maps to see.

Probably its not right to add all discovered tiles to the total tiles as this will screw up the count if not all tiles are built.

Further investigation showed that by default building all maps does not actually build all maps. The config skipJunkMaps is true by default and this means that not all tiles are built by default which is possibly the issue with the percentage for building all maps.

What is adding into the mix is that the amount of total tiles increases as we build tiles, so it may be possible for the percentage to go up and down.

So in short: The calculation of total tiles has to be changed to actually count only tiles that are actually going to be built

**Comment by jackpoz** [comment]
The original PR didn't take into account all this for sure :P

**Comment by jackpoz** [comment]
@Rochet2 can this be closed ?

**Comment by Rochet2** [comment]
yes.
